spec/services/govuk_bank_holiday_retriever_spec.rb

```diff
@@ -2,83 +2,111 @@
 
 RSpec.describe GovukBankHolidayRetriever do
   describe ".dates" do
-    context "successful call" do
-      before do
-        stub_request(:get, "https://www.gov.uk/bank-holidays.json")
-          .to_return(body: json_response, status: 200)
-      end
+    let(:bank_holiday_stub) do
+      stub_request(:get, "https://www.gov.uk/bank-holidays.json").to_return(body: json_response.to_json, status: 200) # .then.to_raise(StandardError)
+    end
+    let(:bank_holiday_stub_again) do
+      stub_request(:get, "https://www.gov.uk/bank-holidays.json").to_return(body: json_response.to_json, status: 200)
+    end
+
+    before do
+      bank_holiday_stub
+      Rails.cache.clear
+    end
```

Comment:
I think we need to leave some notes here for future developers to explain what's going on in this test

Reply:
All specs are fixed and self-explanatory, so no notes required 😄